ddtrace/llmobs/decorators.py

```diff
@@ -9,14 +10,30 @@
 
 def llm(model_name, model_provider=None, name=None, session_id=None):
     def inner(func):
+        span_name = name
+        if span_name is None:
+            span_name = func.__name__
+
+        @wraps(func)
+        def gen_wrapper(*args, **kwargs):
+            if not LLMObs.enabled or LLMObs._instance is None:
+                log.warning("LLMObs.llm() cannot be used while LLMObs is disabled.")
+                for ret in func(*args, **kwargs):
+                    yield ret
```

Comment:
If this branch is hit, then it looks like the generator will be called twice. Is this intentional? Also I think a generator wrapper needs to handle `StopIteration` `GeneratorExit`, `throw` etc...

Reply:
Thanks for pointing this out!